merger.js

```diff
@@ -49,28 +49,32 @@ function processDir(dir)
 }
 
 function processFile(file, include) {
-  if (file === mainFile && mainIsProcessed) {
+  let processedOnce = false;
+  for (let i = 0; i < processOnce.length; i++) {
+    if (processOnce[i] == file) {
+      // console.log("Found in processedOnce: ", file);
+      processedOnce = true;
+      break;
+    }
+  }
+ 
+ if (file == outputFile) return;
+
+ if (file === mainFile && mainIsProcessed) {
     return; //Main can be processed on its own at the start
   } else if (path.extname(file) == ".hpp" && !include) {
     return;
   } else if (path.extname(file) == ".cpp") {
     console.log('Processing ' + file);
     fs.appendFileSync(outputFile, "/*-- File: " + file + " start --*/\n");
-  } else if (path.extname(file) == ".hpp" || path.extname(file) == ".h") {
+  } else if ((path.extname(file) == ".hpp" || path.extname(file) == ".h") && (!processedOnce)) {
```

Comment:
I am not sure why this generated comment should be removed. Initially I had the idea to use these comments later on to be able to breakdown the merged file back into the original files.
I do understand that it generates quite a lot of output, but since that ugly merged file is only used to compile on CodinGame side, I do not see a big issue with it.

Reply:
Ooops. Don't know why I even put this here, sorry.